core/src/main/java/org/ehcache/core/internal/service/ServiceLocator.java

```diff
@@ -438,11 +457,22 @@ ServiceMap lookupDependenciesOf(ServiceMap resolved, Class<? extends Service> re
   private static Set<Class<? extends Service>> identifyTransitiveDependenciesOf(final Class<?> clazz) {
 
     Set<Class<? extends Service>> dependencies = identifyImmediateDependenciesOf(clazz);
+    for (Class<? extends Service> dep : dependencies) {
```

Comment:
We already have circular dependency checking in the startup logic... I understand this fails slightly earlier, but I also think it's going to miss some instances.  The circular dependencies we care about are between implementations... I think this logic will miss things if a service implementation introduces extra links by implementing two service interfaces? Right?

Reply:
Are you sure ? The circular dependency test was ignored and was failing...
I agree it might miss some things and could call stop() on an implementation id the interface is not found as a dependency. But well... It would requires a bigger code change to do it correctly. This is only support the general use case of 1 service 1 impl at least.